pom.xml

```diff
@@ -178,6 +178,9 @@
     <test.java.home>${java.home}</test.java.home>
     <test.exclude.tags></test.exclude.tags>
 
+    <!-- Package to use when relocating shaded classes. -->
+    <spark.shade.packageName>org.spark_project</spark.shade.packageName>
```

Comment:
For total avoidance of trouble, is `org.sparkproject` safer? or are you sure underscores are more OK than hyphens? BTW you could consider this PR as a resolution for SPARK-8815 too.